### PR TITLE
Milestone 2:  Decorate Classes with  composition

### PR DESCRIPTION
**Pull request by RichardChileya**
### In this section, I: 

- [ ] Added a Namable Class
  -  Implemented correct_name_method
- [ ] Inherited Person from Nameable
- [ ] Added CapitalizeDecorator and TrimmerDecorator
   - Created a class that inherits from the base Decorator class.
   - Created a class that inherits from the base Decorator class.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
+require_relative './nameable'
+
+class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
 
-  def initialize(age, name = 'unknown', parrent_permission: true)
+  def initialize(age, name: 'unknown', parrent_permission: true)
+    super()
+
```
Comment (ShadyShawkat) [inline person.rb:9]: - [x] Please make sure that the `Person` class has a `correct_name` method implemented, as per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#:~:text=Make%20sure%20that%20this%20class%20has%20a%20method%20correct_name%20implemented.%20It%20should%20simply%20return%20the%20name%20attribute.) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @RichardChileya, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (RichardChileya) [reply person.rb:9]: thank you. this has been resolved

**Review by prolajumokeoni — APPROVED** [review]
Hi @RichardChileya 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀v
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**